src/Php73/bootstrap.php

```diff
@@ -13,4 +13,12 @@
     if (!function_exists('is_countable')) {
         function is_countable($var) { return is_array($var) || $var instanceof Countable || $var instanceof ResourceBundle || $var instanceof SimpleXmlElement; }
     }
+
+    if (!function_exists('array_key_first')) {
+        function array_key_first(array $array) { reset($array); return key($array); }
```

Comment:
If i'm correct this will change the internal state of the array, while the php 7.3 functionality will keep the internal stat intact.

How about changing this function to not mess with the internal state as well?

Reply:
that's indeed necessary to emulate the correct behavior.

Reply:
[`array_slice`](http://php.net/manual/en/function.array-slice.php) with preserved keys should do the trick

Reply:
well, actually, this will not mess the external array. But it will trigger a copy of the array (to be able to alter the one passed to the function).
So we are emulating the right behavior, but with a performance cost. This should rather make a foreach loop with a return statement in the first iteration.

Reply:
I went for `array_slice` in both cases (passing the array by reference to prevent a copy).
This works as expected and some initial tests shows the internal pointer of the array isn't modified and there is no additional memory overhead

Reply:
no need to pass the array by reference just to avoid a copy. That could actually *create* a copy, and that would not polyfill the PHP 7.3 function, as it would have a different signature